Add Tiny AE live preview

Sakura-Luna opened the pull request:
Added [Tiny AE](https://github.com/madebyollin/taesd) for better live preview.

Here's an example, output and approximation:
![拷贝](https://github.com/AUTOMATIC1111/stable-diffusion-webui/assets/53183413/f763bddb-f683-4e25-b4b5-73138661d614)

Most of the work is done, but there are still some issues:
1. The effect is slightly worse than the original repository's example (color difference), and I don't know if it can be improved.
2. It seems that some details need to be improved.
3. Since its effect and performance are very good, I think Approx NN can be removed.

Sakura-Luna committed:
Add Tiny AE live preview

Sakura-Luna committed:
Add live preview mode check

Sakura-Luna commented:
@AUTOMATIC1111 What do you think about it?

Sakura-Luna commented:
The simple test takes time, Approx is about 0.003s, and Tiny is about 0.016s, which has little effect on the speed.

papuSpartan commented:
Probably should be based against dev branch?

Sakura-Luna commented:
> Probably should be based against dev branch?

They're not very different, and it's not a priority yet.

EfourC commented on modules/sd_samplers_common.py:
```diff
@@ -22,21 +22,26 @@ def setup_img2img_steps(p, steps=None):
+        x_sample = torch.clamp((x_sample * 0.25) + 0.5, 0, 1)
     else:
-        x_sample = processing.decode_first_stage(shared.sd_model, sample.unsqueeze(0))[0]
+        if approximation == 3:
+            x_sample = sd_vae_approx.cheap_approximation(sample)
+        elif approximation == 2:
+            x_sample = sd_vae_approx.model()(sample.to(devices.device, devices.dtype).unsqueeze(0))[0].detach()
+        else:
+            x_sample = processing.decode_first_stage(shared.sd_model, sample.unsqueeze(0))[0]
+        x_sample = torch.clamp((x_sample + 1.0) / 2.0, min=0.0, max=1.0)
```
Why nested and reordered value checks?  Seems cleaner to keep the original structure like:
```
if approximation == 3:
elif approximation == 2:
elif approximation == 1:
else:
```

Sakura-Luna replied:
You ignored line 43, it's not suitable for all cases, layering the `if` is necessary.

EfourC replied:
Ah of course! Sorry about my mistake

Sakura-Luna committed:
Minor changes

Sakura-Luna committed:
Minor change

AUTOMATIC1111 committed:
add credits

AUTOMATIC1111 committed:
return live preview defaults to how they were
only download TAESD model when it's needed
return calculations in single_sample_to_image to just if/elif/elif blocks
keep taesd model in its own directory

AUTOMATIC1111 committed:
Merge branch 'dev' into taesd-a
